src/lib/tls/tls12/msg_client_kex.cpp

```diff
@@ -318,6 +327,20 @@ Client_Key_Exchange::Client_Key_Exchange(const std::vector<uint8_t>& contents,
             if(ka_key->algo_name() == "DH")
                shared_secret = CT::strip_leading_zeros(shared_secret);
 
```

Comment:
Unrelated note: Above is a hard-coded key exchange. This should probably use `Callbacks::tls_ephemeral_key_agreement()` as of #3322 😨. I'll look into this separately.

Reply:
See: https://github.com/randombit/botan/pull/3539